roles/monitoring/tasks/main.yml

```diff
@@ -1,126 +1,135 @@
-- name: "Install snmpd"
+---
+- name: Install snmpd
   ansible.builtin.apt:
-    update_cache: yes
+    update_cache: true
     name: snmpd
     state: present
 
-- name: "Retrieve SNMPv3 password and passphrase"
+- name: Retrieve SNMPv3 password and passphrase from Vault
+  community.hashi_vault.vault_kv2_get:
+    path: infrastructure/mercure
+  register: monitoring_secrets
+  run_once: true
+  delegate_to: localhost
+  become: false
+
+- name: Set SNMPv3 password and passphrase to facts
   ansible.builtin.set_fact:
-    snmp_authpass: "{{ lookup('community.general.hashi_vault', 'secret=secret/data/infrastructure/mercure:snmp_authpass') }}"
-    snmp_cryptopass: "{{ lookup('community.general.hashi_vault', 'secret=secret/data/infrastructure/mercure:snmp_cryptopass') }}"
-    librenms_api_token: "{{ lookup('community.general.hashi_vault', 'secret=secret/data/infrastructure/mercure:librenms_api_token') }}"
+    monitoring_snmp_authpass: "{{ monitoring_secrets.secret.snmp_authpass }}"
+    monitoring_snmp_cryptopass: "{{ monitoring_secrets.secret.snmp_cryptopass }}"
+    monitoring_librenms_api_token: "{{ monitoring_secrets.secret.librenms_api_token }}"
   run_once: true
   delegate_to: localhost
-  become: no
+  become: false
 
 - name: Gather facts using SNMP version 3
   community.general.snmp_facts:
-    host: '{{ inventory_hostname }}'
+    host: "{{ inventory_hostname }}"
     version: v3
     level: authPriv
     integrity: sha
     privacy: aes
-    username: '{{ snmp_user }}'
-    authkey: '{{ snmp_authpass }}'
-    privkey: '{{ snmp_cryptopass }}'
-  ignore_errors: yes
+    username: "{{ monitoring_snmp_user }}"
+    authkey: "{{ monitoring_snmp_authpass }}"
+    privkey: "{{ monitoring_snmp_cryptopass }}"
+  ignore_errors: true
   delegate_to: localhost
-  become: no
+  become: false
   register: snmp_facts_results
 
-- name: "Configure SNMP"
+- name: Configure SNMP
   ansible.builtin.template:
     src: snmpd.conf.j2
     dest: /etc/snmp/snmpd.conf
     owner: root
     group: root
     mode: "0600"
-    backup: yes
-  notify: restart snmpd
+    backup: true
+  notify: Restart snmpd
 
 # Script downloaded from https://github.com/librenms/librenms-agent/blob/master/snmp/distro
-- name: "Copy distro script from LibreNMS"
+- name: Copy distro script from LibreNMS
   ansible.builtin.copy:
     src: distro
     dest: /usr/bin/distro
     owner: root
     group: root
     mode: "0755"
-  notify: restart snmpd
+  notify: Restart snmpd
 
 - name: Setup SNMP v3 user
+  when: snmp_facts_results.failed
+
```

Comment:
extra line return (needs to check the lint settings)

Reply:
I checked with Ansible-lint, when the `when` is at the end of the block it wants to place it above (in line [with their rule](https://ansible.readthedocs.io/projects/lint/rules/key-order/#why-i-no-longer-can-put-when-after-a-block)), but it adds an empty line for no reason (and doesn't complain if you manually remove the line).